api/handlers/eventvalidation.go

```diff
@@ -0,0 +1,149 @@
+		return err
+	}
+	if !slices.Contains(allowedActions, action) {
+		return &EventValidationError{Msg: "unknown event type action: " + action}
+	}
+	if _, ok := validators[action]; !ok {
+		return nil
+	}
+	return validators[action](event)
+
+}
+
+// validateTaskStartedEvent contains logic to validate "sh.keptn.event.task.started" events
+func validateTaskStartedEvent(e models.KeptnContextExtendedCE) error {
+	var eventData v0_2_0.EventData
+	if err := v0_2_0.Decode(e.Data, &eventData); err != nil {
+		return &EventValidationError{Msg: "could not parse common event data"}
+	}
+	if eventData.Project == "" || eventData.Stage == "" || eventData.Service == "" {
+		return &EventValidationError{Msg: "mandatory field(s) 'project', 'stage' or 'service' missing"}
+	}
+
+	if e.Triggeredid == "" {
+		return &EventValidationError{Msg: "mandatroy field 'triggeredid' missing"}
+	}
+	return nil
+}
+
+// validateTaskFinishedEvent contains logic that validates "sh.keptn.event.task.finished" events
+func validateTaskFinishedEvent(e models.KeptnContextExtendedCE) error {
+	var eventData v0_2_0.EventData
+	if err := v0_2_0.Decode(e.Data, &eventData); err != nil {
+		return &EventValidationError{Msg: "could not parse common event data"}
+	}
+	if eventData.Project == "" || eventData.Stage == "" || eventData.Service == "" {
+		return &EventValidationError{Msg: "mandatory field(s) 'project', 'stage' or 'service' missing"}
+	}
+	if eventData.Result == "" {
+		return &EventValidationError{Msg: "result field is not set"}
+	}
+	if eventData.Status == "" {
+		return &EventValidationError{Msg: "status field is not set"}
+	}
+
+	if e.Triggeredid == "" {
+		return &EventValidationError{Msg: "mandatroy field 'triggeredid' missing"}
+	}
+	return nil
+}
+
+// validateSequenceTriggeredEvent contains logic that validates "sh.keptn.dev.sequence.triggered" events
+func validateSequenceTriggeredEvent(e models.KeptnContextExtendedCE) error {
+	var eventData v0_2_0.EventData
+	if err := v0_2_0.Decode(e.Data, &eventData); err != nil {
+		return &EventValidationError{Msg: "could not parse common event data"}
+	}
+	if eventData.Project == "" || eventData.Stage == "" || eventData.Service == "" {
+		return &EventValidationError{Msg: "mandatory field(s) 'project', 'stage' or 'service' missing"}
+	}
+	stage, _, _, err := v0_2_0.ParseSequenceEventType(*e.Type)
+	if err != nil {
+		return &EventValidationError{Msg: "unknown event type"}
+	}
+	if eventData.Stage != stage {
+		return &EventValidationError{Msg: "stage name in event data and in event type does not match"}
+	}
+	return nil
+}
+
+func validateMonitoringConfigureEvent(e models.KeptnContextExtendedCE) error {
+	type data struct {
+		v0_2_0.EventData
+		Type string `json:"type,omitempty"`
+	}
+
+	var eventData data
+	if err := v0_2_0.Decode(e.Data, &eventData); err != nil {
+		return &EventValidationError{Msg: "could not parse common event data"}
+	}
+
+	if eventData.Project == "" || eventData.Service == "" || eventData.Type == "" {
```

Comment:
i believe only the project name is mandatory for the monitoring.configure event, so we should remove the check for the service property